class/Defaults/Trigger/Plugin/PluginTrigger.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Plugin trigger abstract
+ *
+ * @package notification
+ */
+
+namespace BracketSpace\Notification\Defaults\Trigger\Plugin;
+
+use BracketSpace\Notification\Abstracts;
+use BracketSpace\Notification\Defaults\MergeTag;
+
+/**
+ * Post trigger class
+ */
+abstract class PluginTrigger extends Abstracts\Trigger {
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $params trigger configuration params.
+	 */
+	public function __construct( $params = array() ) {
```

Comment:
You can get rid of this entire method if you are just passing params. So the child classes would pass two params instead of 1 array